README.md
## 项目指南



### 环境
JDK：1.8以上(java8)，不能用1.9 (Java9)  
IDE：默认使用Eclipse  或 IntelliJ IDEA

### 运行
当直接下载代码，用eclipse打开运行前，第一次请先执行命令：

> clean package

软件的各种资源文件（组态，配置，网页）需要能在开发状态和打包状态下都能正确取到。所以，增加了BasePath类来判断环境是开发还是运行状态来进行基准目录的修正。  

开发状态：软件的运行目录为：nurse\target\classes  
运行状态：软件的运行目录为：nurse当前目录  

因为，当你从git上下载代码后，没有target目录，所以必需先打包一次，才能正确访问组态画面等资源文件。   
如果你需要修改测试资源文件，请每次执行打包命令，否则和IDE的内容不同步  

之后，执行NurseApp作为应用启动，就可以看到程序正常运行。

### 打包
目前的项目是Maven工程，建立run config, 设置nurse为工作目录，Goals内容如下：  

> clean package

也可以直接右键运行Run As的Maven Build，也可以直接运行。  

打包后将在target目录产生nurse-1.0.1目录，所有执行所需要的文件都已经放好在这个目录下，将这个目录拷贝到任何目录都可以运行。




CHANGELOG.md
# Change Log
所有重要的改动都要在这个文档中加以记录。

## [Unreleased]

### Changed


### Fixed


### Deprecated
- 历史曲线表仍有测试数据，正式发布前需去掉

## [3.3.4] - 2020-04-09
### Modify
- iView 设置页面优化


## [3.3.3] - 2020-04-03
### Modify
- 组态页面引用，隐藏导入导出
- 重启添加清理脚本
- 告警分布整合到历史记录中
- iView版本首页与设备列表页分离


## [3.3.2] - 2020-04-01
### Added
- 异常登录限制
### Modify
- 登录记住密码优化
- B接口-联通配置文件优化
- 提示框位置优化
- 协议包导出优化
- 修改完mainconfig.xml配置后重新加载最新的配置
- 表格组态 告警时信号值变色

## [3.3.1] - 2020-03-21
### Added
- 组态包导出，入口：配置 => 组态管理
### Modify
- 模板管理：导出协议完整版 导出协议模板xml与so


## [3.2.2] - 2020-02-27
### Added
- 设备信息功能，入口：组态页面的设备详情，右下角i号



## [3.2.1] - 2020-01-19
### Added
- 模板管理：导出协议 v0.5 导出协议模板xml
### Modify
- 组态转非组态页面优化


## [3.2.0]  - 2020-01-15
### Added
- 组态转静态页面功能


## [3.1.5] - 2019-12-24
- 将定制项目有用的功能整合到标准版中
### Added
- 添加更适合触屏的设置页面
- 游客权限添加告警阀值设置功能
### Modify
- 时间控件优化
- 触屏样式优化
- 确定弹出框模态化


## [3.1.2] - 2019-12-11
### Added
- 刷新遮蔽页面，解决刷新页面加载丑问题
### Modify
- alarm提示框样式优化
- U盘升级iView状态页面
- 配置栏在管理员和游客不同权限不同选择
- 优化64节温场


## [3.1.1] - 2019-12-07
### Modify
- iView虚拟键盘优化： 弹出键盘优化
- iVIew虚机键盘优化： 多出输入框添加虚拟键盘
- 设备详情优化： 固定表头
- 64节温场优化： 添加机柜图和温场定位配置功能
- 实时告警组态优化： 将时间放到第二行，优化在iView上时间换行问题



## [3.1.0] - 2019-11-28
### Added
- 64节温场
### Modify
- 优化组态页上面设备栏选择效果
- 历史图表组态添加平均值、最大值和最小值选择
- 解决告警联动翻译问题
- 解决时间加载异常问题



## [3.0.3] - 2019-11-22
### Added
- 实时告警组态
### Modify
- 修改License生效条件，为删除factor_stat文件生效
- 解决海康威视实时预览有时候404问题
- 解决删除协议时删除掉SO问题
- 解决修改标题失败问题，分离mainconfig.xml的DataHandlerConfig
- 优化虚拟组态计算公式
- 优化iView历史数据查询



## [3.0.2] - 2019-11-14
### Modify
- 接口配置-移动B接口配置化
- 接口配置-联通B接口配置化
- 接口配置-SNMP接口配置化


## [3.0.1] - 2019-11-08
### Added
- 免登陆，mainconfig.xml的<loginPassword>输入正确密码
- 实时图表组态：新仪表盘



## [2.11.7] - 2019-10-29
### Added
- 根据“自诊断设备”的硬盘占用率清理系统文件和数据库数据
- 微模块历史数据组态页
### Fixed
- 解决iView环境license注册问题


## [2.11.6] - 2019-10-18
### Fixed
- 解决：46的1、“类型”框大小；3、在屏上隐藏“导出”按钮
- 解决：48优化配置管理，新增设备的端口类型选择和修改设备重新绑定设备模板
- 解决：49新增协议的“通讯状态”事件没有基类编号问题
- 解决：51实时告警页面，最后一页优化
- 解决：53组态编辑框的删除按钮样式
- 告警通知筛选接收者列表
- 添加一键添加自诊断设备



## [2.11.5] - 2019-10-09
### Added
- 组态-实时图表 ADD:平均值、最大值、最小值
- 组态-实时组态 ADD:风格
- 组态 - ADD:3D MDC
### Fixed
- [组态]优化组态缓存
- 1、去掉柱形图帅选；2、解决实时图表组态选择设备问题
- 优化组态-超链接
- 组态-实时图表 柱形图：隐藏Y轴、添加柱子颜色配置
- 组态-历史图表 优化Y轴
- MDC历史数据 Demo 解决加载异常问题
- 历史告警记录，没有数据时，提示“Not Data”,英文版改用“No data”,中文版改用“没有数据”
- 模板管理相关样式优化
- 遥控遥调组态的密码选择，无密码、登录密码和共用密码


## [2.11.4] - 2019-09-27
- [文本组态]优化信号选择状态
- [MDC配置]机柜温湿度存储过程优化
- [风格切换]优化切换图片不存在问题
- [实时图表]添加柱形图图表
- [历史告警]优化iView历史告警查询
- [组态]添加信号状态 & [表格页面]信号通讯异常灰色、告警红色变化
- [超链接组态]添加组态页面跳转&返回按钮
- [配置-模板管理]屏蔽事件
- [MDC配置-门控制]添加环境设备的控制功能



## [2.11.3] - 2019-09-12
- [组态页面配置]删除设备后并删除设备分类的父级
- 遥调的数值控制，编辑框的值最好取关联信号的值，取不到再按现在的逻辑第一次加0.5，以后加1
- 【图表组态】加载慢
- 实时图表控制的格式
- [文本组态]添加信号选择状态



## [2.11.2] - 2019-08-30
### Added
- [组态]添加小键盘，方便在屏上移动、拉伸、选择组态
- [iView]内置浏览器看门狗
### Fixed
- [组态]“系统概况->冷通道折线图”中，绑定信号选择时，有两个本机IO
- [组态]实时图表控制的格式，单位
- [组态]给所有组态输入框添加了虚拟键盘
- [告警弹出框]白色风格优化


## [2.11.1] - 2019-08-23
### Fixed
- 放大虚拟键盘
- 修改获取IP方式，添加IView的兼容


## [2.11.0] - 2019-08-15
### Added
- [系统风格]实现风格切换



## [2.10.2] - 2019-08-01
### Added
- [门禁记录]保留最后一次刷卡记录
### Fixed
- [修改设备]显示关联的模板
- [邮件告警]发件时间 时差
- [移动B接口]优化配置文件


## [2.10.0] - 2019-07-31
### Added
- [iView 3.0]移植Linux平板开发
- [关于]添加二维码图
- [MainConfig.XML]添加免登陆功能
### Fixed
- [配置]解决修改设备配置影响同端口设备问题
- [自动备份还原配置]具体请看EquipmentProvider.java


## [2.9.2] - 2019-06-29
### Added
- [组态]仪表盘联控：以仪表盘显示数据变化，并下发控制
### Fixed
- [组态]虚拟组态：添加更多样式选择


## [2.9.1] - 2019-06-13
### Added
- [组态]能量柱组态：根据表达式显示百分比，并根据值下发控制


## [2.8.8] - 2019-06-03
### Added
- [B接口]集成中国联通B接口
- [系统操作]恢复访问账户
### Fixed
- [B接口]优化B接口功能


## [2.8.6] - 2019-05-29
### Added
- [B接口]打印配置内容
### Fixed
- [B接口]解决读写中文乱码问题


## [2.8.5] - 2019-05-28
### Added 
- [B接口]移动B接口配置和报文

## [2.8.2] - 2019-05-09
### Fixed
- [协议管理]添加删除协议时so是否在其他协议中存在判断
- [数据管理]添加删除提示
- [MDC系统概况]MDC首页，中间的安全监控等，修改数量时修改最右边的圆角样式
- [资产管理]优化一键生成的位置信息
- [组态页面]优化页面跳转，解决空组态页面


## [2.8.0] - 2019-4-26
### Added 
- [中英文]添加中英文页面切换功能


## [2.7.1] - 2019-04-13
### Added
- [遥控遥调组态]添加控制联动功能，实现群控/延迟控制


## [2.7.0] - 2019-03-22
### Added
- [机架资产管理]高级配置 => 资产机架管理
- 1、目前功能：读取数码人资产条上报的资产信息，与拆除后的下架信息
			

## [2.6.2] - 2019-03-18
### Added
- [信号历史图表]点击“信号”组态、“表格”组态和表格页面的信号，弹出信号的7天历史曲线

### Fixed
- [历史信号缓存]原取一天的平均值，现在为一天的最大值，觉得比较有意义
- [系统操作]关机和重启后退出到登录页面
	
	
## [2.6.1] - 2019-02-22
### Added
- [萤石云视频]使用萤石云服务器平台直接实时预览摄像头
- 1、 特点：摄像头部署在萤石云服务器上(改服务器使用萤石云的服务器，不是我们自己的)和网速有关
- 2、 入口：配置 => MDC配置 => 点击摄像头(或冷通道方格) => 设备类型为“萤石云直播”,新增 => RTMP输入：rtmp://rtmp.open.ys7.com/openlive/e6e0a477c0ca45b29638d27364da1c5f.hd(该rtmp是公司的摄像头部署到外网的直播连接) => 保存
- [表格组态]新开发一个表格页面的组态
- 1、 入口：高级配置 => 组态页面配置 => 选择“子节点”，修改 => 勾选“表格化” => 保存，最后到组态页面

### Fixed
- [组态]表格组态，添加无边框的风格
- [微调]logo的样式
		

## [2.6.0] - 2019-01-16
### Added
- [数码人资产条]资产柜开发(暂停，等待需求明确再启动)
		
		
## [2.5.4] - 2018-12-29
### Fixed
- Web连接java的连接数超过100条重启采集器（解决连接数超过100无法登陆Nurse问题）
- [门禁管理]门禁返回判断
- [门禁管理]门控制，不在门规制表里，默认CmdToken,Value
- [门禁管理]门禁添加门控制类型
- [门禁管理]解决TBL_DoorCard的CardId为-1的问题
- [门禁管理][门禁]加卡，不关联的门不下发时间组命令


## [2.5.3] - 2018-12-21
### Fixed
- [配置生效]添加加载中
- [MDC]冷通道样式优化
- 异常处理
- 解决门禁反馈处理异常
		
		
## [2.5.2] - 2018-12-07
### Fixed
- [MDC配置]通道设备配置化
- [系统概况]动态加载通道设备
- [组态]虚拟组态添加含义


## [2.5.1] - 2018-11-19
### Fixed
- [备份配置]备份上传的图片；备份MainConfig.xml
	
	
## [2.4.3] - 2018-11-10
### Fixed
- [组态页面配置]隐藏也会在设备列表中
- [组态]优化虚拟组态


## [2.4.2] - 2018-11-02
### Added
- [操作记录]用户操作日志；历史 => 操作记录（添加了TBL_UserOperationLog表）
### Fixed
- [配置]备份备注优化
- [配置文件mainconfig.xml]添加告警提示音替换(true：语音播报；false：语音提示)
		
		
## [2.4.1] - 2018-10-26
### Fixed
- [系统概况]环境量显示
- [MDC配置]机柜U位管理
- [网络电话]添加定时保平安功能（修改了TBL_NetworkPhone表结构）
- [告警通知]样式优化
- [表达式]添加AND和OR选项
### Added
- [导航]添加导航指引功能；告警配置=>导航
- [系统操作]添加关机和重启功能；关机：关闭MySQL和Nurse的服务，不断电十分钟后重启；重启：就是reboot
		
		
## [2.4.0] - 2018-10-19
### Fixed
- [刷卡记录]添加打印查库时间日志
- [门禁规制]门禁控制根据门禁规制的格式
- [License]优化
- [门禁管理]门管理添加删除所有时间组和卡
- [门禁管理-时限管理]优化双门单门时限下发
- [门禁管理-门管理]添加删除所有时间组
- [告警联动]解决触发表达式的长度问题


## [2.3.4] - 2018-09-30
### Fixed
- [MDC系统概况]优化机柜的资产信息
- [资产管理]资产编号为唯一键
- [资产管理]修改资产编码生成规则
- [资产管理]分页、筛选优化；导出Xls优化
- [页面时间]一小时定时校准页面时间
- [门禁管理]根据单双门筛选时段限制
- [门禁管理]优化快捷加卡读取卡号
- \#IN3LO:【资产管理】导出，文件格式错误

web/js/login.js
//debugger;

(function() {
    getLogoAndTitle();
	if (!window.JSON) {
		window.JSON = {
			parse: function(sJSON) {
				return eval('(' + sJSON + ')');
			},
			stringify: (function() {
				var toString = Object.prototype.toString;
				var isArray = Array.isArray || function(a) {
					return toString.call(a) === '[object Array]';
				};
				var escMap = {
					'"': '\\"',
					'\\': '\\\\',
					'\b': '\\b',
					'\f': '\\f',
					'\n': '\\n',
					'\r': '\\r',
					'\t': '\\t'
				};
				var escFunc = function(m) {
					return escMap[m] || '\\u' + (m.charCodeAt(0) + 0x10000).toString(16).substr(1);
				};
				var escRE = /[\\"\u0000-\u001F\u2028\u2029]/g;
				return function stringify(value) {
					if (value === null) {
						return 'null';
					} else if (typeof value === 'number') {
						return isFinite(value) ? value.toString() : 'null';
					} else if (typeof value === 'boolean') {
						return value.toString();
					} else if (typeof value === 'object') {
						if (typeof value.toJSON === 'function') {
							return stringify(value.toJSON());
						} else if (isArray(value)) {
							var res = '[';
							for (var i = 0; i < value.length; i++)
								res += (i ? ', ' : '') + stringify(value[i]);
							return res + ']';
						} else if (toString.call(value) === '[object Object]') {
							var tmp = [];
							for (var k in value) {
								if (value.hasOwnProperty(k))
									tmp.push(stringify(k) + ': ' + stringify(value[k]));
							}
							return '{' + tmp.join(', ') + '}';
						}
					}
					return '"' + value.toString().replace(escRE, escFunc) + '"';
				};
			})()
		};
	}

	$(".input-group input").on('keydown', function (e) {
		var key = e.which;
		if (key == 13) {
			e.preventDefault();
			$("#btnLogin").click();
		}
	});

	$('#LoginId').click(function(event){
		if($("#LoginId").get(0).checked){
			var username = $("#username").val();
			var pwd = $("#pwd").val();
			if(username != "" && pwd != "")
				sessionStorage.setItem("UserNameAndPwd",username+"|"+baseEncode(username +"|"+pwd));
		}else{
			$("#pwd").val("");
			sessionStorage.setItem("UserNameAndPwd","");
		}
	});

	$("#btnLogin").click(function(event) {
		$("#showtime").remove();

        if(Remain == 0 || Remain == "0"){
			var language = JSON.parse(sessionStorage.getItem("languageJson"));
            $("#errText").text(language.Login.License.Prompt);/*"试用期结束，请点击右下角注册！"*/
            return;
        }

		var dataObj = {};
		var dataAarray = [];
		var username = $("#username").val();
		var pwd = $("#pwd").val();

		var dataValue = baseEncode(username +"|"+pwd);

		if($("#LoginId").get(0).checked){
			var split = ["",""];
			if(sessionStorage.getItem("UserNameAndPwd") != undefined)
				split = sessionStorage.getItem("UserNameAndPwd").split("|");
			//存储的用户名和密码与记录的用户名和密码相同
			if(username == split[0] && pwd == split[1]){
				dataValue = split[1];
				sessionStorage.setItem("UserNameAndPwd",username+"|"+dataValue);
			}else
				sessionStorage.setItem("UserNameAndPwd",username+"|"+baseEncode(username +"|"+pwd));
		}

		dataAarray.push({
			K: "user.login",
			V: dataValue
		});

		dataObj.data = dataAarray;

		var encoded = JSON.stringify(dataObj);

		$.post('/data', encoded, function(data, textStatus, xhr) {
			var resObj = JSON.parse(data);
			var res = resObj[0].V;

			if (res.indexOf("OK") != -1){
				var result = res.split("|");
				localStorage.setItem("username",username);
				localStorage.setItem("token",result[1]);
                localStorage.setItem("versions",result[2]);
                localStorage.setItem("isAdmin",result[3]);
				localStorage.setItem("needLogin","true");
                var loginHome = result[4];
				localStorage.setItem("systemStyle",result[5]);
				//上一次登入记录
				localStorage.setItem("LastLoginDate",result[6]);
				localStorage.setItem("LastLoginIP",result[7]);
				//自动登出时间
				localStorage.setItem("logoutTime",result[8]);

                localStorage.setItem("showHelp",true);
                if(loginHome && isNaN(loginHome)){
					if(result[2] == "IView")
						$(window.location).attr("href", "iview.html" + loginHome);
					else
						$(window.location).attr("href", "index.html" + loginHome);

					localStorage.setItem("loginHome",loginHome);
				}else{
					if(result[2] == "Room")
						$(window.location).attr("href", "index.html#/adevice/1004/adiagram");
					else if(result[2] == "2D")
						$(window.location).attr("href", "index.html#/mdcalarm");
					else if(result[2] == "3D")
						$(window.location).attr("href", "mdc.html");
					else
						$(window.location).attr("href", "iview.html#/adevice/8890/adiagram");
				}

			}else{
				var prompt = languageLogin.Prompt;

				var maxError = 0;
				if(res.indexOf("|") > -1) {
					var split = res.split("|");
					res = split[0];
					maxError = split[1];
				}

                if(res == "Password Error"){//密码错误
                    result = prompt.PasswordError;//"密码错误";
					initLoginLimit(maxError,languageLogin.Limit);
                }else if(res == "Username And Password Not Entered"){//未输入帐户和密码
                    result = prompt.NotEntered;//"未输入用户名和密码";
                }else if(res == "Enter Username And Password Error"){//输入帐户和密码错误
                    result = prompt.EnterError;//"输入的用户名和密码错误";
                }else if(res == "DataBase Connection Failed"){//数据库连接失败
                    result = prompt.DataBaseFailed;//"数据库连接失败";
                }else if(res == "Overdue"){
                    result = prompt.Overdue;//"用户已过期";
                }
				$("#errText").text(result);
			}
		});
        //删除Cookie
        setCookie('activeText',event.currentTarget.outerText,-1);
        function setCookie(c_name,value,expiredays)
        {
            var exdate=new Date();
            exdate.setDate(exdate.getDate()+expiredays);
            document.cookie=c_name+ "=" +escape(value)+
                ((expiredays==null) ? "" : ";expires="+exdate.toGMTString());
        }
	});

	//region 登录限制
	// login页面初始化 或 点击登入 密码错误 时执行此函数
	function initLoginLimit(maxError,limit){
		if(limit == undefined)
			limit = JSON.parse(sessionStorage.getItem('languageJson')).Login.Limit;

		//如果maxError 为 undefined则表示为 login 页面初始化
		if(maxError == undefined){
			var dataObj = {};
			var dataArray = [];
			dataArray.push({
				K: "user.getMaxError",
				V:""
			});
			dataObj.data = dataArray;

			var encoded = JSON.stringify(dataObj);
			$.post('/data', encoded, function(data, textStatus, xhr) {
				var numError = JSON.parse(data)
				var str = numError[0].V;
				if(str == undefined || str == "null") return;
				maxError = str;
				timeTransition(maxError,limit.Abnormal,limit.Unlock);
			});
			return;
		}
		timeTransition(maxError,limit.Abnormal,limit.Unlock);
	}
	// 时间转换
	function timeTransition(maxError,abnormal,unlock) {
		abnormal = abnormal.replace(/\[Count\]/ig, maxError);
		switch (parseInt(maxError)) {
			case 5:
				var endTime = new Date(new Date().getTime()+1000*60);
				jqueryAlert({
					'content' : '<div style="font-size: 30px">'+ abnormal+
						'<div id="showtime"></div> </div>',
					'closeTime' : 1000*60,
					'modal'        : true,
					'minWidth'     : '450'
				});
				break;
			case 6:
				var endTime = new Date(new Date().getTime()+1000*60*5);
				jqueryAlert({
					'content' : '<div style="font-size: 30px">' + abnormal+
						'<div id="showtime"></div></div> ',
					'closeTime' : 1000*60*5,
					'modal'        : true,
					'minWidth'     : '450'
				});
				break;
			case 7:
				var endTime = new Date(new Date().getTime()+1000*60*15);
				jqueryAlert({
					'content' : '<div style="font-size: 30px">' + abnormal+
						'<div id="showtime"></div></div> ',
					'closeTime' : 1000*60*15,
					'modal'        : true,
					'minWidth'     : '450'
				});
				break;
			case 8:
				var endTime = new Date(new Date().getTime()+1000*60*30);

				jqueryAlert({
					'content' : '<div style="font-size: 30px">' + abnormal+
						'<div id="showtime"></div></div> ',
					'closeTime' : 1000*60*30,
					'modal'        : true,
					'minWidth'     : '450'
				});
				break;
			case 9:
				var endTime = new Date(new Date().getTime()+1000*60*60);
				jqueryAlert({
					'content' : '<div style="font-size: 30px">' + abnormal+
						'<div id="showtime"></div> </div>',
					'closeTime' : 1000*60*60,
					'modal'        : true,
					'minWidth'     : '450'
				});
				break;
			case 10:
				var endTime = "永久";
				jqueryAlert({
					'content' : ' <div style="font-size: 30px">' + abnormal+
						'<div id="showtime"></div> </div>',
					'closeTime' : 1000*5,
					'modal'        : true,
					'minWidth'     : '450'
				});
				break;
		}

		//若 numError为5或以上则每秒都执行一次该方法
		if(maxError >= 5){
			//延迟一秒执行自己
			var interval = setInterval(function(){
				var bool= countTime(endTime,unlock);
				if(bool == true){
					clearInterval(interval);
				}
			},1000);
		}

	}

	function countTime(endTime,unlock) {
		if("永久"== endTime)return;
		//开始时间
		var nowDate = new Date();
		//结束时间
		//相差的总秒数
		var totalSeconds = parseInt((endTime - nowDate) / 1000);

		//取模（余数）
		var modulo = totalSeconds % (60 * 60 * 24);
		modulo = modulo % (60 * 60);
		//分钟
		var minutes = Math.floor(modulo / 60);
		//秒
		var seconds = modulo % 60;

		if( minutes <= 0 && seconds <= 0){
			return true;
		}
		//输出到页面
		unlock = unlock.replace(/\[m\]/ig, minutes);
		unlock = unlock.replace(/\[s\]/ig, seconds);
		//$("#showtime").html("解锁倒计时:"  + minutes + "分钟" + seconds + "秒");
		$("#showtime").html(unlock);

	}
	//endregion

    function getLogoAndTitle(){
        var dataObj = {};
        var dataArray = [];
        dataArray.push({
            K: "user.getTitleAndLogo",
            V: baseEncode("userTitle|userLogo")
        });
        dataObj.data = dataArray;

        var encoded = JSON.stringify(dataObj);
        $.post('/data', encoded, function(data, textStatus, xhr) {
            var resObj = JSON.parse(data);
            var res = resObj[0].V;
            var userTitle = res.split("|")[0];
            var userLogo = res.split("|")[1];
            localStorage.clear();
            localStorage.setItem("userTitle",userTitle);
            localStorage.setItem("userLogo",userLogo);
            $(document).attr("title",userTitle);
            $(".logotitle").text(userTitle);
            $(".logo").attr("src",userLogo);
        });

        if(sessionStorage.getItem("UserNameAndPwd") != undefined && sessionStorage.getItem("UserNameAndPwd")!= ""){
			var unap = sessionStorage.getItem("UserNameAndPwd").split("|");
            $("#username").val(unap[0]);
			$("#pwd").val(unap[1]);
            $("#LoginId").get(0).checked = true;
        }
    }


    function baseEncode(str){
		var c1, c2, c3;
		var base64EncodeChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
		var i = 0;
		var len= str.length;
		var string = '';

		while (i < len){
			c1 = str.charCodeAt(i++) & 0xff;
			if (i == len){
				string += base64EncodeChars.charAt(c1 >> 2);
				string += base64EncodeChars.charAt((c1 & 0x3) << 4);
				string += "==";
				break;
			}
			c2 = str.charCodeAt(i++);
			if (i == len){
				string += base64EncodeChars.charAt(c1 >> 2);
				string += base64EncodeChars.charAt(((c1 & 0x3) << 4) | ((c2 & 0xF0) >> 4));
				string += base64EncodeChars.charAt((c2 & 0xF) << 2);
				string += "=";
				break;
			}
			c3 = str.charCodeAt(i++);
			string += base64EncodeChars.charAt(c1 >> 2);
			string += base64EncodeChars.charAt(((c1 & 0x3) << 4) | ((c2 & 0xF0) >> 4));
			string += base64EncodeChars.charAt(((c2 & 0xF) << 2) | ((c3 & 0xC0) >> 6));
			string += base64EncodeChars.charAt(c3 & 0x3F)
		}
		return string;
	}

    function getCookie(c_name){
        if (document.cookie.length>0)
        {
            c_start=document.cookie.indexOf(c_name + "=");
            if (c_start!=-1)
            {
                c_start=c_start + c_name.length+1;
                c_end=document.cookie.indexOf(";",c_start);
                if (c_end==-1) c_end=document.cookie.length;
                return unescape(document.cookie.substring(c_start,c_end));
            }
        }
        return ""
    }

    var Remain = 999;
    function checkoutLicense() {
        var dataObj = {};
        var dataAarray = [];
        dataAarray.push({
            K: "license.CheckoutLicense",
            V: ""
        });

        dataObj.data = dataAarray;

        var encoded = JSON.stringify(dataObj);

        $.post('/data', encoded, function(data, textStatus, xhr){
            var resObj = JSON.parse(data);
            resObj = JSON.parse(resObj[0].V);
            if(resObj.IsShow == false || resObj.IsShow == "false"){
                $("#btnLicense").hide();
            }else{
				//linux系统隐藏注册按钮
				if(window.navigator.userAgent.indexOf("Windows") == -1)
					$("#btnLicense").hide();
				else {
					$("#btnLicense").show();
					$("#btnLicense span").html(resObj.Remain);
				}
                Remain = resObj.Remain;
            }
        });
    }

	var languageLogin = undefined;
    //中英文切换
	function getLanguage(){
		var dataObj = {};
		var dataAarray = [];
		dataAarray.push({
			K: "language.GetLoginLanguageJson",
			V: ""
		});
		dataObj.data = dataAarray;

		var encoded = JSON.stringify(dataObj);
		$.post('/data', encoded, function(data, textStatus, xhr){
			var resObj = JSON.parse(data);
			var json = JSON.parse(resObj[0].V);

			languageLogin = json.Login;
			parseValue(json);

			setLanguageSession(json.Language);
		});
	}
	getLanguage();

	function parseValue(json){
		//登录
		$("#username").attr("placeholder",json.Login.UserName);
		$("#pwd").attr("placeholder",json.Login.Password);
		$("#hint").html(json.Login.Hint);
		$("#btnLogin").html(json.Login.Login);
		$("#btnLicense").html(json.Login.Probation);
		//风格
		$("#ui-style").html(json.Login.Style.Title);
		$("#ui-blue").html(json.Login.Style.SkyBlue);
		$("#ui-white").html(json.Login.Style.TechnologyWhite);
		//语言
		$("#language").val(json.Language);
		//License
		$("#Register-License").html(json.Login.License.Title);
		$(".Generate-License").html(json.Login.License.GenerateLicense);
		$("#Upload-License").html(json.Login.License.UploadLicense);
		$("#Upload-License-Button").html(json.Login.License.UploadLicenseButton);

		checkoutLicense();
	}

	$("#language").change(function(){
		var language = $('#language option:selected').val();

		var dataObj = {};
		var dataAarray = [];
		dataAarray.push({
			K: "language.SetLanguage",
			V: language
		});
		dataObj.data = dataAarray;

		var encoded = JSON.stringify(dataObj);
		$.post('/data', encoded, function(data, textStatus, xhr){
			var resObj = JSON.parse(data);
			var json = JSON.parse(resObj[0].V);
			parseValue(json);

			setLanguageSession(json.Language);
		});
	});

	//存储中英文到Session
	function setLanguageSession(language){
		var url = undefined;
		if(language == "Chinese") url = "../data/language/ch.json";
		else url = "../data/language/en.json";
		$.getJSON(url,function(data){
			sessionStorage.setItem('languageJson.language', data.Language);
			sessionStorage.setItem("languageJson",JSON.stringify(data));

			initLoginLimit(undefined,data.Login.Limit);
		});
	}

	//是否需要登录
	function needLogin(){
		var dataObj = {};
		var dataAarray = [];
		dataAarray.push({
			K: "user.needLogin",
			V: ""
		});
		dataObj.data = dataAarray;

		var encoded = JSON.stringify(dataObj);
		$.post('/data', encoded, function(data, textStatus, xhr){
			var resObj = JSON.parse(data);
			var res = resObj[0].V;
			if(res != "TRUE"){
				var result = res.split("|");
				localStorage.setItem("versions",result[0]);
				localStorage.setItem("isAdmin",result[1]);
				localStorage.setItem("needLogin","false");
				localStorage.setItem("systemStyle",result[3]);
				localStorage.setItem("username",result[4]);
				var loginHome = result[2];

				localStorage.setItem("showHelp",true);
				if(loginHome && isNaN(loginHome)){
					$(window.location).attr("href", "index.html"+loginHome);
					localStorage.setItem("loginHome",loginHome);
				}else{
					if(result[0] == "Room")
						$(window.location).attr("href", "index.html#/adevice/1004/adiagram");
					else if(result[0] == "2D")
						$(window.location).attr("href", "index.html#/mdcalarm");
					else if(result[0] == "3D")
						$(window.location).attr("href", "mdc.html");
					else
						$(window.location).attr("href", "iview.html#/adevice/8890/adiagram");
				}
			}
		});
	}
	needLogin();

	/* 风格切换 */
	$("#systemStyle").change(function(){
		var oldStyle = $('#systemStyle option:selected').val();

		var dataObj = {};
		var dataAarray = [];
		dataAarray.push({
			K: "systemStyle.SetSystemStyle",
			V: oldStyle
		});
		dataObj.data = dataAarray;

		var encoded = JSON.stringify(dataObj);
		$.post('/data', encoded, function(data, textStatus, xhr){
			var resObj = JSON.parse(data);
			var newStyle = resObj[0].V;

			setStyle(newStyle);

			location.reload();
		});
	});

	function setStyle(style){
		//系统风格
		$("#systemStyle").val(style);

		var random = parseInt(10000*Math.random());
		if(style != "Blue")
			$("#StyleLink").attr("href", "css/"+style+"Style.css?v=" + random);
		else
			$("#VersionLink").attr("href","css/versions2.css?v="+random);

        $(".loginbody").addClass("Body-Background");
        $(".login-bg").addClass("Login-Background");
	}

	function getSystemStyle(){
		var dataObj = {};
		var dataAarray = [];
		dataAarray.push({
			K: "systemStyle.GetSystemStyle",
			V: ""
		});
		dataObj.data = dataAarray;

		var encoded = JSON.stringify(dataObj);
		$.post('/data', encoded, function(data, textStatus, xhr){
			var resObj = JSON.parse(data);
			var newStyle = resObj[0].V;

			setStyle(newStyle);
		});
	}
	getSystemStyle();
})();